geoh5py/objects/object_base.py

```diff
@@ -430,44 +503,63 @@ def last_focus(self, value: str):
 
         self._last_focus = value
 
-    def mask_by_extent(
-        self, extent: np.ndarray, inverse: bool = False
-    ) -> np.ndarray | None:
+    def load_children_values(self):
+        """
+        Load the values of the children in memory.
         """
-        Sub-class extension of :func:`~geoh5py.shared.entity.Entity.mask_by_extent`.
+        for child in self.children:
+            _ = getattr(child, "values", None)
 
-        Applied to object's centroids.
+    @property
+    def locations(self) -> np.ndarray | None:
         """
+        Exposes the vertices or centroids of the object.
+        """
+        out = None
+        if hasattr(self, "vertices"):
+            out = self.vertices
+        if hasattr(self, "centroids"):
+            out = self.centroids
+
+        return out
+
+    def mask_by_extent(
+        self, extent: np.ndarray, inverse: bool = False
+    ) -> np.ndarray | None:
         if self.extent is None or not box_intersect(self.extent, extent):
             return None
 
         return mask_by_extent(self.locations, extent, inverse=inverse)
 
     @property
-    def n_cells(self):
+    def n_cells(self) -> int:
```

Comment:
```suggestion
    @abstractmethod
    def n_cells(self) -> int:
```

Maybe we should turn those as @abstracmethod, and force the sub-classes to implement it.

Reply:
I simply removed them, has it does not make sense to have a grid with vertices.